TShockAPI/Handlers/SendTileSquareHandler.cs

```diff
@@ -0,0 +1,445 @@
+				args.Player.SendTileSquare(tileX, tileY, size);
+				return;
+			}
+
+			if (args.Player.IsBeingDisabled())
+			{
+				TShock.Log.ConsoleDebug("Bouncer / SendTileSquare rejected from being disabled from {0}", args.Player.Name);
+				args.Player.SendTileSquare(tileX, tileY, size);
+				return;
+			}
+
+			bool[,] processed = new bool[size, size];
+			NetTile[,] tiles = new NetTile[size, size];
+			for (int x = 0; x < size; x++)
+			{
+				for (int y = 0; y < size; y++)
+				{
+					tiles[x, y] = new NetTile(args.Data);
+				}
+			}
+
+			Debug.VisualiseTileSetDiff(tileX, tileY, size, size, tiles);
+
+			for (int x = 0; x < size; x++)
+			{
+				for (int y = 0; y < size; y++)
+				{
+					// Do not process already processed tiles
+					if (processed[x, y])
+					{
+						continue;
+					}
+
+					int realx = tileX + x;
+					int realy = tileY + y;
+
+					// Do not process tiles outside of the world boundaries
+					if ((realx < 0 || realx >= Main.maxTilesX)
+						|| (realy < 0 || realy > Main.maxTilesY))
+					{
+						processed[x, y] = true;
+						continue;
+					}
+
+					// Do not process tiles that the player cannot update
+					if (!args.Player.HasBuildPermission(realx, realy) ||
+						!args.Player.IsInRange(realx, realy))
+					{
+						continue;
+					}
+
+					NetTile newTile = tiles[x, y];
+					TileObjectData data;
+
+					// If the new tile has an associated TileObjectData object, we take the tile and the surrounding tiles that make up the tile object
+					// and process them as a tile object
+					if (newTile.Type < TileObjectData._data.Count && (data = TileObjectData._data[newTile.Type]) != null)
+					{
+						NetTile[,] newTiles = new NetTile[data.Width, data.Height];
+						for (int i = 0; i < data.Width; i++)
+						{
+							for (int j = 0; j < data.Height; j++)
+							{
+								newTiles[i, j] = tiles[x + i, y + j];
+								processed[x + i, y + j] = true;
+							}
+						}
+
+						ProcessTileObject(newTile.Type, data, newTiles, realx, realy, args);
+						continue;
+					}
+
+					// If the new tile does not have an associated tile object, process it as an individual tile
+					ProcessSingleTile(realx, realy, newTile, size, args);
+					processed[x, y] = true;
+				}
+			}
+
+			// Uncommenting this function will send the same tile square 10 blocks above you for visualisation. This will modify your world and overwrite existing blocks.
+			// Use in test worlds only.
+			//Debug.DisplayTileSetInGame(tileX, tileY - 10, size, size, tiles, args.Player);
+
+			// If we are handling this event then we have updated the server's Main.tile state the way we want it.
+			// At this point we should send our state back to the client so they remain in sync with the server
+			if (args.Handled == true)
+			{
+				args.Player.SendTileSquare(tileX, tileY, size);
+				TShock.Log.ConsoleDebug("Bouncer / SendTileSquare reimplemented from spaghetti from {0}", args.Player.Name);
```

Comment:
```suggestion
				TShock.Log.ConsoleDebug("Bouncer / SendTileSquare reimplemented from new greatness from {0}", args.Player.Name);
```

Since the original comment was named after entirely inscrutable spaghetti, this should be different now!